runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -8310,6 +8310,13 @@ void post_fma_and_wait(c_nodeid_t locale, gni_post_descriptor_t* post_desc,
   int cdi;
   atomic_bool post_done;
   uint64_t iters = 0;
+  // Avoid yielding for tasks with limited comm. Avoids artificially increasing
```

Comment:
It might be worthwhile to put `if (do_yield) { }` around this whole thing.  That way we can avoid the task-private-data access overhead for the cases where we're already not going to yield.

Reply:
Yeah, good idea.

FWIW accessing task-local storage is relatively cheap under qthreads -- about twice as fast as uncontested atomic test-and-set lock (so roughly twice as fast as acquiring a comm domain.) I think almost all the cost is function call overhead, and we should be able to optimize that out down the line with link-time-optimization or if qthreads provided TLS-getting through a header.